cli/lib/nf3.mjs

```diff
@@ -661,15 +665,20 @@ class Nf3 {
     };
     connection.onmessage = async message => {
       const msg = JSON.parse(message.data);
-      const { type, txDataToSign } = msg;
+      const { type, txDataToSignList } = msg;
       logger.debug(`Proposer received websocket message of type ${type}`);
       if (type === 'block') {
-        const res = await this.submitTransaction(
-          txDataToSign,
-          this.stateContractAddress,
-          this.BLOCK_STAKE,
-        );
-        newGasBlockEmitter.emit('gascost', res.gasUsed);
+        logger.debug(`Found ${txDataToSignList.length} blocks to process`);
+
+        txDataToSignList.reduce((seq, txDataToSign) => {
```

Comment:
Is the `reduce` method used here to submit transactions sequentially and not to reduce an array to a value?
If so, this is new to me :) I usually use `for..of` loop to do this.

Reply:
Yes, reduce pattern. I learned it from @IlyasRidhuan suggestion and after reviewing some information.

Reply:
Yes the reduce pattern is preferred if we need to sequentially resolve promise. It avoids having to prevent the `eslint` warning about `awaiting in a for loop`

Reply:
`for await...of` can be used for synchronous iteration for an async/sync iterable objects

Reply:
I confirm `for... of` also allows synchronous iteration, didn't know about the `reduce` pattern